pkg/mover/chart.go

```diff
@@ -256,12 +257,68 @@ func (cm *ChartMover) loadChartFromIntermediateBundle(bundlePath string) error {
 // loadChartFromPath load the chart in memory from a given path
 func (cm *ChartMover) loadChartFromPath(path string) error {
 	var err error
-	if cm.chart, err = loader.Load(path); err != nil {
-		return &ChartLoadingError{Path: path, Inner: err}
+	dir := path
+	if isChartFile(path) {
+		// always unpack the chart tarball to a directory to avoid issues loading
+		// files if the tgz contains replicated entries
+		// https://github.com/vmware-tanzu/asset-relocation-tool-for-kubernetes/issues/111
+		if dir, err = expandedChartDir(path); err != nil {
```

Comment:
why don't you just override the path here?

You are not adding any protection by having two variables since you are setting one as the other

Reply:
Usually linters and such frown upon method or function inputs mutated inside. And for good reason, because you maybe changing the invariants on the code in ways that might be not be clear, or even lead to bugs.
- In this method `path` is the input **path** given originally, which _could be a tarball or a directory_.
- On the other hand `dir` is the directory in which files are already expanded, can be original or temporary.
Also this way we never lose track of the original path, when `dir` differs.